experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1246 @@
-            sequence of one cp_object, then the next). CPObject #1 and
-            CPO #2 must have same intt and intn. Integrations will
-            switch between one and the other until time is up/nave is
-            reached.
-        PULSE : Simultaneous sequence interfacing, pulse by pulse
-            creates a single sequence. CPO A and B might have different
-            frequencies (stereo) and/or may have different pulse
-            length, mpinc, sequence, but must have the same integration
-            time. They must also have same len(scan), although they may
-            use different directions in scan. They must have the same
-            scan boundary if any. A time offset between the pulses
-            starting may be set (seq_timer in cp_object). CPObject A
-            and B will have integrations that run at the same time.
-        """
-
-    def __call__(self): # REVIEW #28 It seems this method is used to print stuff. try to use __repr__ or __str__ instead http://stackoverflow.com/questions/1436703/difference-between-str-and-repr-in-python
-        print 'CPID [cpid]: {}'.format(self.cpid)
-        print 'Num of CP Objects [cponum]: {}'.format(self.cponum)
-        for i in range(self.cponum):
-            print '\n'
-            print 'CP Object : {}'.format(i)
-            print self.cpo_list[i]
-        print '\n'
-        print 'Interfacing [interface]: {}'.format(self.interface)
-        return None
+    """ A prototype experiment class composed of metadata, including experiment slices (exp_slice) 
+    which are dictionaries of radar parameters. Basic, traditional experiments will be composed 
+    of a single slice. More complicated experiments will be composed of multiple slices that 
+    interface in one of four pre-determined ways, as described under interface_types.
+    
+    This class is used via inheritance to create experiments.
+    
+    Some variables shouldn't be changed by the experiment, and their properties do not have setters. 
+    Some variables can be changed in the init of your experiment, and can also be modified 
+    in-experiment by the class method 'update' in your experiment class. These variables have been
+    given property setters.
+    """
+
+    __slice_keys = ["slice_id", "cpid", "txantennas", "rx_main_antennas",
+                    "rx_int_antennas", "pulse_sequence", "pulse_shift", "mpinc",
+                    "pulse_len", "nrang", "frang", "intt", "intn", "beam_angle",
+                    "beam_order", "scanboundflag", "scanbound", "txfreq", "rxfreq",
+                    "clrfrqrange", "acf", "xcf", "acfint", "wavetype", "seqtimer"]
+
+    __hidden_slice_keys = ['rxonly', 'clrfrqflag']
+
+    """ The slice keys are described as follows: 
+    
+    DESCRIPTION OF SLICE KEYS
+
+slice_id : The ID of this object. An experiment can have multiple objects.
+cpid: The ID of the experiment, consistent with existing radar control programs.
```

Comment:
No I don't think so. Consider twofsound, it would have two slices (two frequencies interfaced scan by scan). Also, we don't want to rule out a 'smart experiment' in the future that would adjust its own frequency or some other parameter and be constantly changing its own slices. I think such an experiment should be given one cpid.

Reply:
Is that what you meant by this comment? Confused by the quotes.